### ISSUE TITLE
SQLiteDatabaseLockedException

### ISSUE THREAD

**Issue by kkroid** [renefloor/flutter_cache_manager#79]
2019-04-16 15:29:25.240 12156-12156/com.my.test E/SQLiteDatabase: Failed to open database '/data/user/0/com.my.test/databases/test_image_cache.db'.
    android.database.sqlite.SQLiteDatabaseLockedException: database is locked (code 5): , while compiling: PRAGMA journal_mode
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:890)
        at android.database.sqlite.SQLiteConnection.executeForString(SQLiteConnection.java:635)
        at android.database.sqlite.SQLiteConnection.setJournalMode(SQLiteConnection.java:321)
        at android.database.sqlite.SQLiteConnection.setWalModeFromConfiguration(SQLiteConnection.java:295)
        at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:216)
        at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:194)
        at android.database.sqlite.SQLiteConnectionPool.openConnectionLocked(SQLiteConnectionPool.java:493)
        at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:200)
        at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:192)
        at android.database.sqlite.SQLiteDatabase.openInner(SQLiteDatabase.java:864)
        at android.database.sqlite.SQLiteDatabase.open(SQLiteDatabase.java:849)
        at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:750)
        at android.database.sqlite.SQLiteDatabase.openOrCreateDatabase(SQLiteDatabase.java:767)
        at com.tekartik.sqflite.SqflitePlugin$Database.open(SqflitePlugin.java:792)
        at com.tekartik.sqflite.SqflitePlugin$Database.access$900(SqflitePlugin.java:783)
        at com.tekartik.sqflite.SqflitePlugin.onOpenDatabaseCall(SqflitePlugin.java:667)
        at com.tekartik.sqflite.SqflitePlugin.onMethodCall(SqflitePlugin.java:762)
        at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:201)
        at io.flutter.view.FlutterNativeView$PlatformMessageHandlerImpl.handleMessageFromDart(FlutterNativeView.java:188)
        at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:202)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:325)
        at android.os.Looper.loop(Looper.java:142)
        at android.app.ActivityThread.main(ActivityThread.java:6553)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

**Comment by kkroid** [comment]
It happens when I quickly tap back button and relaunch my app again

**Comment by kkroid** [comment]
you may need update sqflite plugin to the latest version

**Comment by The-Redhat** [comment]
Same error is showing up in my app.

**Comment by renefloor** [comment]
Should be fixed in version 1.0.0 and higher. If not please reopen.

**Comment by mhmzdev** [comment]
Facing the same issue with version `3.3.1` via firebase crashlytics for some users.

```
android.database.sqlite.SQLiteConnection.nativePrepareStatement (SQLiteConnection.java)
android.database.sqlite.SQLiteConnection.acquirePreparedStatement (SQLiteConnection.java:1047)
android.database.sqlite.SQLiteConnection.executeForString (SQLiteConnection.java:790)
android.database.sqlite.SQLiteConnection.setJournalMode (SQLiteConnection.java:407)
android.database.sqlite.SQLiteConnection.setWalModeFromConfiguration (SQLiteConnection.java:337)
android.database.sqlite.SQLiteConnection.open (SQLiteConnection.java:260)
android.database.sqlite.SQLiteConnection.open (SQLiteConnection.java:205)
android.database.sqlite.SQLiteConnectionPool.openConnectionLocked (SQLiteConnectionPool.java:505)
android.database.sqlite.SQLiteConnectionPool.open (SQLiteConnectionPool.java:206)
android.database.sqlite.SQLiteConnectionPool.open (SQLiteConnectionPool.java:198)
android.database.sqlite.SQLiteDatabase.openInner (SQLiteDatabase.java:928)
android.database.sqlite.SQLiteDatabase.open (SQLiteDatabase.java:908)
android.database.sqlite.SQLiteDatabase.openDatabase (SQLiteDatabase.java:772)
android.database.sqlite.SQLiteDatabase.openDatabase (SQLiteDatabase.java:761)
android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked (SQLiteOpenHelper.java:373)
android.database.sqlite.SQLiteOpenHelper.getWritableDatabase (SQLiteOpenHelper.java:316)
androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase (FrameworkSQLiteOpenHelper.java:4)
androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase (FrameworkSQLiteOpenHelper.java:4)
androidx.room.RoomDatabase.inTransaction (RoomDatabase.java:2)
androidx.room.RoomDatabase.assertNotSuspendingTransaction (RoomDatabase.java)
androidx.work.impl.model.SystemIdInfoDao_Impl.getWorkSpecIds (SystemIdInfoDao_Impl.java:9)
androidx.work.impl.background.systemjob.SystemJobScheduler.reconcileJobs (SystemJobScheduler.java:20)
androidx.work.impl.utils.ForceStopRunnable.cleanUp (ForceStopRunnable.java:11)
androidx.work.impl.utils.ForceStopRunnable.forceStopRunnable (ForceStopRunnable.java)
androidx.work.impl.utils.ForceStopRunnable.run (ForceStopRunnable.java:31)
androidx.work.impl.utils.SerialExecutor$Task.run (SerialExecutor.java:2)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1137)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:637)
java.lang.Thread.run (Thread.java:1012)
``` 

cc: @renefloor
